src/util/requestDispatcher.js

```diff
@@ -110,8 +142,9 @@ export default class RequestDispatcher {
       _arp.skipResARP = true
       return addToURL(url, 'arp', compressData(JSON.stringify(_arp), this.logger))
     }
-    if (StorageManager._isLocalStorageSupported() && typeof localStorage.getItem(ARP_COOKIE) !== 'undefined' && localStorage.getItem(ARP_COOKIE) !== null) {
```

Comment:
These checks are not required. `readFromLSorCookie` already checks for `_isLocalStorageSupported` supported. 
i created a variable which gets the value from LS and checks if it is undefined or not.